hadoop-hdds/docs/content/feature/Short-Circuit-Read.md

```diff
@@ -0,0 +1,76 @@
+
+The Hadoop native library "libhadoop.so" provides the support to use the Unix domain socket. Please refer to Native Libraries ("https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-common/NativeLibraries.html") for details of this library.
+
+Before enabling short-circuit local reads, you must have a proper libhadoop.so, and make sure it's under the directory where Java can find and load it through "System.loadLibrary()" call.
+
+The paths that Java will search for libraries are specified by the "java.library.path" property. The default value of "java.library.path" depends on the OS and Java version. For example, on Linux with OpenJDK 8 it is `/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib`.
+
+Command "ozone checknative" can be used to detect whether libhadoop.so can be loaded successfully by Ozone service. 
+
+## Configuration
+
+Short-circuit local reads need to be configured on both the DataNode and the client. By default, it is disabled.
+
+```XML
+<property>
+   <name>ozone.client.read.short-circuit</name>
+   <value>false</value>
+   <description>Disable or enable the short-circuit local read feature.</description>
+</property>
+```
+
+It makes use of a UNIX domain socket, a special path in the filesystem. You will need to set a path to this socket. 
+
+```XML
+<property>
+   <name>ozone.domain.socket.path</name>
+   <value>/var/lib/ozone_dn_socket</value>
+   <description>The path used to create domain socket.</description>
+</property>
+```
+
+The DataNode needs to be able to create this path. On the other hand, it should not be possible for any user except the Ozone user(user who launches Ozone service) or root to create this path. For this reason, paths under /var/run or /var/lib are often used， just like the current default value "/var/lib/ozone_dn_socket".
+
+If you configure the "ozone.domain.socket.path" to other value, for example "/dir1/dir2/ozone_dn_socket", please make sure that both dir1 and dir2 are exiting directories, and there is no ozone_dn_socket under dir2. ozone_dn_socket will be created by Ozone Datanode later during Datanode start.
+
+### Security Consideration
+
+To ensure data security and integrity, Ozone will follow the rule as Hadoop ("https://cwiki.apache.org/confluence/display/HADOOP2/SocketPathSecurity") on "ozone.domain.socket.path" path permission check. 
+It will fail the "ozone.domain.socket.path" verification and disable the feature if the filesystem permissions of the specified path are inadequate. 
+The verification failure message carries detail instruction about how to fix the problem. Following is an example, 
+
+"The path component: '/etc/hadoop' in '/etc/hadoop/ozone_dn_socket' has permissions 0777 uid 0 and gid 0. It is not protected because it is world-writable. This might help: 'chmod o-w /etc/hadoop'. For more information: https://wiki.apache.org/hadoop/SocketPathSecurity"
```

Comment:
```suggestion
The verification failure message carries instructions about how to fix the problem. Here is an example: 

\```
The path component: '/etc/hadoop' in '/etc/hadoop/ozone_dn_socket' has permissions 0777 uid 0 and gid 0. It is not protected because it is world-writable. This might help: 'chmod o-w /etc/hadoop'. For more information: https://wiki.apache.org/hadoop/SocketPathSecurity
\```
```